### ISSUE TITLE
Call assignments are created with no default "Done" criteria (`goal_filters`)

### ISSUE THREAD

**Issue by richardolsson** [zetkin/app.zetkin.org#1418]
## Description
When creating a new call assignment, it should be configured such by default that people are only spoken to once in that assignment. This is usually done by including a `call_history` filter in the `goal_filters`. But by default right now, the `goal_filters` is empty, meaning that there is no definition of done and a person can basically be spoken to an infinite number of times in the same assignment.

## Steps to reproduce
1. Go to http://localhost:3000/organize/1/projects/1
2. Click on "Create activity"
3. Click "Create call assignment"
4. Click "Edit done criteria"

### Expected Behaviour
Should show a Smart Search with a filter to find all people who have been reached in the current call assignment.

### Actual Behaviour
Shows an empty Smart Search

### Empty done criteria (Smart Search)
![image](https://github.com/zetkin/app.zetkin.org/assets/550212/c645bf21-be59-4d49-a573-e389abbc4aca)

## Proposed solution
After creating the Call Assignment (in `CampaignActionButtons`), immediately patch the `goal_filters` array, with the ID of the newly created assignment.

```tsx
goal_filters = [{
  type: 'call_history',
  config: {
    operator: 'reached',
    assignment: '<id>', // <--- put ID of freshly created assignment here
  }
}];
```

**Comment by andreas-karlsson** [comment]
@richardolsson I'll take a stab at this one if no one else has started yet.

**Comment by niklasva82** [comment]
This is blocked because $self in campaign/{campaign_id}/callassignments does not work properly on the backend.

**Comment by richardolsson** [comment]
That's a pity! Honestly I can't believe we (that is, I myself) even implemented `$self` in the first place, because I think it's quite ugly. The solution I would have gone with today is to create the call assignment first, then get it's ID and patch it (in an RPC). So that's another solution that would work to solve this issue.

**Comment by richardolsson** [comment]
> The solution I would have gone with today is to create the call assignment first, then get it's ID and patch it (in an RPC). So that's another solution that would work to solve this issue.

I've updated the issue to reflect this proposed solution instead, and changed the tag as this does make the issue slightly bigger (but it's still a small medium). 😊

**Comment by richardolsson** [comment]
Resolved by #2093
